internal/pkg/cli/deploy/env.go

```diff
@@ -53,6 +67,34 @@ func (d *envDeployer) UploadArtifacts() (map[string]string, error) {
 	return urls, nil
 }
 
+// DeployEnvironment deploys an environment using CloudFormation.
+func (d *envDeployer) DeployEnvironment(in *deployEnvironmentInput) error {
```

Comment:
from the method name it feels like it should do either CreateAndRender or UpdateAndRender depending on whether the stack exists or not - did you have something else in mind?

Reply:
The workflow in my mind is:
1. `env init` does `Create` only. It should error out if the env stack already exists, because `env init` is expected to run against only new environments.
2. `env deploy` does `Update` only. It should and will error out if the env stack does not exist. The stack should have been created by `env init`.

This `DeployEnvironment` method is meant to do the latter "update", and I wanted to define another method for the former that's typically called by `env init`:
```
func (d *envDeployer) BootstrapEnvironment() {
    ...
    input := &deploy.CreateEnvironmentInput {
        Name: d.env.Name,
       App: deploy.AppInformation{
			Name:                d.app.Name,
			Domain:              d.app.Domain,
			AccountPrincipalARN: in.rootUserARN,
		},
		AdditionalTags:       d.app.Tags,
     }
    d.envDeployer.CreateAndRenderEnvironment(os.Stderr, input)
}
```